diff --git a/utils/aperf.c b/utils/aperf.c
index 8ecff2c..785d0ed 100644
--- a/utils/aperf.c
+++ b/utils/aperf.c
@@ -145,7 +145,7 @@ static int read_msr(int cpu, unsigned int idx, unsigned long long *val)
 		return -1;
 	if (lseek(fd, idx, SEEK_CUR) == -1)
 		goto err;
-	if (read(fd, val, sizeof val) != sizeof *val)
+	if (read(fd, val, sizeof *val) != sizeof *val)
 		goto err;
 	close(fd);
 	return 0;
